### PR TITLE
Add a few links for JupyterHub on Hadoop

### PR DESCRIPTION
**Pull request by jcrist**
This adds links in the docs to a few projects for deploying JupyterHub on a Hadoop cluster.

- Deploying JupyterHub on Hadoop Guide: https://github.com/jcrist/jupyterhub-on-hadoop
- YarnSpawner: https://github.com/jcrist/yarnspawner
- KerberosAuthenticator: https://github.com/jcrist/kerberosauthenticator

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,7 +150,7 @@ To start the Hub on a specific url and port ``10.0.1.2:443`` with **https**:
 | PAMAuthenticator                                                            | Default, built-in authenticator                   |
 | [OAuthenticator](https://github.com/jupyterhub/oauthenticator)              | OAuth + JupyterHub Authenticator = OAuthenticator |
 | [ldapauthenticator](https://github.com/jupyterhub/ldapauthenticator)        | Simple LDAP Authenticator Plugin for JupyterHub   |
-| [kdcAuthenticator](https://github.com/bloomberg/jupyterhub-kdcauthenticator)| Kerberos Authenticator Plugin for JupyterHub      |
+| [kerberosauthenticator](https://github.com/jcrist/kerberosauthenticator)    | Kerberos Authenticator Plugin for JupyterHub      |
```
Comment (jcrist) [inline README.md]: I feel a bit bad about replacing an existing authenticator, but the bloomberg one was broken, untested, and unmaintained (I reached out to ask about contributing and got no response). This one is tested and works on current JupyterHub. Happy to revert this change if needed.

File: docs/source/gallery-jhub-deployments.md [hunk 2]
```diff
@@ -160,6 +160,10 @@ easy to do with RStudio too.
 - https://getcarina.com/blog/learning-how-to-whale/
 - http://carolynvanslyck.com/talk/carina/jupyterhub/#/
 
+### Hadoop
+
+- [Deploying JupyterHub on Hadoop](https://jcrist.github.io/jupyterhub-on-hadoop/)
```
Comment (jcrist) [inline docs/source/gallery-jhub-deployments.md]: This is written in the style of `zero-to-jupyterhub-k8s` but for Hadoop clusters. As it's seen less testing than TLJH and zero-to-jupyterhub I figured it'd be best to put it in the gallery instead of in the distributions list. Not sure what's best.

File: README.md (hunk 1 above)
Comment (minrk) [reply README.md]: No worries, it's good to know this!

File: docs/source/gallery-jhub-deployments.md (hunk 2 above)
Comment (minrk) [reply docs/source/gallery-jhub-deployments.md]: Makes sense to me. If you feel it would be better moved over later, feel free to come on back.

File: docs/source/reference/spawners.md [hunk 3]
```diff
@@ -25,6 +25,8 @@ Some examples include:
   run without being root, by spawning an intermediate process via `sudo`
 - [BatchSpawner](https://github.com/jupyterhub/batchspawner) for spawning remote
   servers using batch systems
+- [YarnSpawner](https://github.com/jcrist/yarnspawner) for spawning notebook
```
Comment (minrk) [inline docs/source/reference/spawners.md]: If you would like to maintain this on the jupyterhub org, I'd be happy to move it over and preserve your ownership. No worries if not.

**Comment by minrk** [comment]
This is great, thanks!
